File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/ghost/ButtonValues.java
package org.firstinspires.ftc.teamcode.util.ghost;

import java.util.ArrayList;
import java.util.Arrays;

public class ButtonValues extends ControllerValues<Boolean> {
  public final static String buttonA="a";
  public final static String buttonB="b";
  public final static String buttonX="x";
  public final static String buttonY="y";
  public final static String dpadUp="du";
  public final static String dpadDown="dd";
  public final static String dpadLeft="dl";
  public final static String dpadRight="dr";

  public final static String bumperLeft="bl";
  public final static String bumperRight="br";

  public final static String joystickLeft="jl";
  public final static String joystickRight="jr";

  public final static String start="st";
  public final static String options="op";


  public ButtonValues() {
    super(false,
            new ArrayList<Boolean>(Arrays.asList(new Boolean[]{false,false,false,false,false,false,false,false,false,false})),
            new ArrayList<String>(Arrays.asList(new String[]{buttonA,buttonB,buttonX,buttonY,dpadUp,dpadDown,dpadLeft,dpadRight,bumperLeft,bumperRight}))
    );

  }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/ghost/ButtonValues.kt
package org.firstinspires.ftc.teamcode.util.ghost

class ButtonValues() : ControllerValues<Boolean>(
        false,
        arrayListOf(false, false, false, false, false, false, false, false, false, false),
        arrayListOf(buttonA, buttonB, buttonX, buttonY, dpadUp, dpadDown, dpadLeft, dpadRight, leftBumper, rightBumper)
) {
    companion object {
        const val buttonA = "a"
        const val buttonB = "b"
        const val buttonX = "x"
        const val buttonY = "y"
        const val dpadUp = "du"
        const val dpadDown = "dd"
        const val dpadLeft = "dl"
        const val dpadRight = "dr"
        const val leftBumper = "lb"
        const val rightBumper = "rb"
    }

}